blueflood-core/src/test/java/com/rackspacecloud/blueflood/cache/TokenCacheTest.java

```diff
@@ -0,0 +1,162 @@
+
+    @Before
+    public void setUp() {
+        cache.resetCache();
+    }
+
+    @Test
+    public void basicStuffWorks() {
+        List<Token> tokens = Token.getTokens(
+                Locator.createLocatorFromPathComponents("12345", "com", "example", "a", "b", "c"));
+        for (Token token : tokens) {
+            assertThat(cache.isTokenCurrent(token), is(false));
+            cache.setTokenCurrent(token);
+            assertThat(cache.isTokenCurrent(token), is(true));
+        }
+        // Five tokens expected: com, com.example, com.example.a, com.example.a.b, and com.example.a.b.c
+        // The tenant isn't considered a separate token, but instead is part of each token's identifier.
+        assertThat(cache.getCurrentLocatorCount(), equalTo(5L));
+    }
+
+    @Test
+    public void resetWorks() {
+        List<Token> tokens = Token.getTokens(Locator.createLocatorFromDbKey("12345.a.b.c"));
+        for (Token token : tokens) {
+            cache.setTokenCurrent(token);
+            assertThat(cache.isTokenCurrent(token), is(true));
+        }
+        assertThat(cache.getCurrentLocatorCount(), equalTo(3L));
+        cache.resetCache();
+        assertThat(cache.getCurrentLocatorCount(), equalTo(0L));
+        for (Token token : tokens) {
+            assertThat(cache.isTokenCurrent(token), is(false));
+        }
+    }
+
+    @Test
+    public void expiresStuff() throws Exception {
+        // Given several random tokens
+        Random r = new Random();
+        List<Token> tokens = new ArrayList<>();
+        for (int i = 0; i < 10; i++) {
+            Locator locator = Locator.createLocatorFromDbKey(
+                    "tenant.l1-" + r.nextInt(10) + ".l2-" + r.nextInt(10) + ".leaf");
+            tokens.addAll(Token.getTokens(locator));
+        }
+        // And a cache with a fairly short timeout
+        TokenCache cache = TokenCache.getInstance(
+                100, MILLISECONDS, 100, MILLISECONDS);
+        // When I insert all the tokens into the cache
+        tokens.forEach(cache::setTokenCurrent);
+        // Then the cache reflects all of the entries
+        tokens.forEach(token -> assertThat(cache.isTokenCurrent(token), is(true)));
+        // And after the timeout, all of them are no longer cached
+        Thread.sleep(101);
+        tokens.forEach(token -> assertThat(cache.isTokenCurrent(token), is(false)));
+    }
+
+    @Test
+    // Repeat in order to exercise multiple values for the various random settings in the test, providing greater
+    // overall coverage.
+    @Repeating(repetition = 25)
+    public void handlesRandomConcurrencyLoad() {
+        Random r = new Random();
+        ConcurrentLinkedQueue<Token> tokensAddedToCache = new ConcurrentLinkedQueue<>();
+        int concurrency = 2 + r.nextInt(20);
+        int executions = 1 + r.nextInt(1000);
+        System.out.println("Test concurrency with " + concurrency + " threads and " + executions + " executions");
```

Comment:
Yes, it's nice to have it there, since it's like running sub-tests within the test. I'd rather it be actual, separate test cases, but I didn't find a convenient way to do that when I wrote this in the LocatorCache test. That test has the same println in it, and I think it works.